prymer/api/variant_lookup.py

```diff
@@ -254,6 +250,30 @@ def __init__(
         self.min_maf: Optional[float] = min_maf
         self.include_missing_mafs: bool = include_missing_mafs
 
+    @classmethod
+    def build(
+        cls,
+        vcf_paths: list[Path],
+        use_cache: Optional[bool] = True,
+        min_maf: Optional[float] = 0.00,
+        include_missing_mafs: Optional[bool] = False,
+    ) -> "VariantLookup":
+        """Constructs a `VariantLookup` object based on size of VCF(s) to query.
+
+        If `use_cache` is True, return a `VariantOverlapDetector` that caches all variants in
+        memory for fast lookup (appropriate for small VCFs). Otherwise, return a
+        `FileBasedVariantLookup` that queries indexed VCFs on disk for each lookup (appropriate
+        for larger VCFs).
+
+        """
+        if use_cache:
+            return VariantOverlapDetector(
+                vcf_paths=vcf_paths, min_maf=min_maf, include_missing_mafs=include_missing_mafs
+            )
+        return FileBasedVariantLookup(
+            vcf_paths=vcf_paths, min_maf=min_maf, include_missing_mafs=include_missing_mafs
+        )
```

Comment:
**suggestion** Since the two classes have the same signature, you could make this a little shorter like so:
```py
        lookup_cls = VariantOverlapDetector if use_cache else FileBasedVariantLookup
        lookup = lookup_cls(vcf_paths=vcf_paths, min_maf=min_maf, include_missing_mafs=include_missing_mafs)
        
        return lookup
```

Otherwise I'd suggest including the `else` branch:
```py
        if use_cache:
            lookup = VariantOverlapDetector(
                vcf_paths=vcf_paths, min_maf=min_maf, include_missing_mafs=include_missing_mafs
            )
        else:
            lookup = FileBasedVariantLookup(
                vcf_paths=vcf_paths, min_maf=min_maf, include_missing_mafs=include_missing_mafs
            )

        return lookup
```

I think I've talked about this on slack, but it's easier to test and reason about when your function has a single return statement. And it's easier to drop a breakpoint and debug if the return statement doesn't also include logic

Reply:
I personally prefer the second (more explicit) option, and agree with @msto on using an `else` branch.